system-reinstall-bootc/src/podman.rs

```diff
@@ -6,7 +6,33 @@ use bootc_utils::CommandRunExt;
 use std::process::Command;
 use which::which;
 
-pub(crate) fn reinstall_command(image: &str, ssh_key_file: &str) -> Command {
+struct BootcVersion {
+    major: u8,
+    minor: u8,
+    patch: u8,
+}
+
+fn bootc_has_clean(image: &str) -> Result<bool> {
+    let output = Command::new("podman")
+        .args(["run", image, "bootc", "--version"])
+        .output()?;
+    let stdout_str = String::from_utf8_lossy(&output.stdout);
+    let version_str = stdout_str
+        .split_whitespace()
+        .nth(1)
+        .ok_or_else(|| anyhow::anyhow!("Failed to parse bootc version"))?;
```

Comment:
I probably wrote https://github.com/bootc-dev/bootc/issues/1312#issuecomment-2874008011 after you started working on this but what do you think about that?

To be clear I don't have a really strong opinion on this, there's clear advantages to parsing the version number too in that in the future it'd let us query it just once and then determine *other* feature flags based on that too.